api/v3/Order.php

```diff
@@ -59,3 +59,42 @@ function civicrm_api3_order_get($params) {
   $params['sequential'] = $isSequential;
   return civicrm_api3_create_success($contributions, $params, 'Order', 'get');
 }
+
+/**
+ * Delete a Order.
+ *
+ * @param array $params
+ *   Input parameters.
+ *
+ * @return array
+ */
+function civicrm_api3_order_delete($params) {
+  $contribution = civicrm_api3('Contribution', 'get', array(
```

Comment:
Just a note here - one way to make this shorter would be 

if (!civicrm_api3('Contribution', 'getvalue', array('return' => 'is_test', 'id' => $params['id')) {
throw new ....
}

Note you can be sure $params['id'] will be set due to your aliasing - you don't need to check contribution_id.

Also - I suggest you edit the exception to say only test orders can be deleted.